graph/telemetry/istio/appender/appender.go

```diff
@@ -316,3 +340,36 @@ func getAppWorkloads(namespace, app, version string, gi *graph.AppenderGlobalInf
 	}
 	return result
 }
+
+func getApp(namespace, appName string, gi *graph.AppenderGlobalInfo) (*models.App, bool) {
+	if appName == "" || appName == graph.Unknown {
+		return nil, false
+	}
+
+	var allAppsMap map[string]appsMap
+	if existingAllAppsMap, ok := gi.Vendor[appsMapKey]; ok {
+		allAppsMap = existingAllAppsMap.(map[string]appsMap)
+	} else {
+		allAppsMap = make(map[string]appsMap)
+		gi.Vendor[appsMapKey] = allAppsMap
+	}
+
+	var namespaceApps appsMap
+	if existingNamespaceApps, ok := allAppsMap[namespace]; ok {
+		if app, ok := existingNamespaceApps[appName]; ok {
+			return app, true
+		} else {
+			namespaceApps = existingNamespaceApps
+		}
+	} else {
+		namespaceApps = appsMap{}
+		allAppsMap[namespace] = namespaceApps
+	}
+
+	if app, err := gi.Business.App.GetApp(context.TODO(), namespace, appName); err == nil {
```

Comment:
A long time ago I remember adding GetServiceDefinition() support, specifically to make it more efficient to just pull a lighter subset of information, perhaps we need something analogous.  Or maybe since then there have been other changes and what I did for GetServiceDefinition() is now dated.  Just mentioning this so that we come up with something consistent.

Reply:
For sure, all we need here is the labels. If we can come up with a more efficient "GetAppDefinition" or whatever, we should do that now.

Reply:
This part was under review, and there were N "GetServiceXYZ" methods that don't have sense.
This code is being consolidated with two main "GetAppList, GetApp" but passing criteria object to populate the result with the info you need (just the definition, the health, validations, runtimes, etc).


Also, I was thinking more to provide these labels in the first pass of the GetApp (or GetAppList, I miss some context, but I haven't studied the changes in the whole context).

Reply:
> Also, I was thinking more to provide these labels in the first pass of the GetApp

We won't know that information at the time the call is made. The true graph filtering is done client side via the cytoscape API.

Reply:
> If you agree I can send a PR to your work with the changes.

Yes. let's use whatever APIs you suggest to keep in sync with the latest business API.

Reply:
@jmazzitelli Perhaps in the graph/telemetry/istio/appender/appender.go we could add a "getAppList" and then the "getApp" would just iterate over the cached list (similar as it does in the "getWorkloadList" and "getServiceList" in the same file).

The "business.GetAppList" would need criteria, but it's something that is also proposed https://github.com/kiali/kiali/pull/4763/files#diff-80818359e8256dc52cb343d2aa947665c308d27ffb25b1930707e115ded5061fR27 as well.

I think that way would be more efficient, than iterating over the GetApp (the GetApp would need in any case criteria as by default it fetches additional things that the graph doesn't need, as the custom dashboards).

Perhaps as there are two PR that may use this AppCriteria, I can add in master a PR with the GetApp criteria, and both branches rebase it (just thinking how to minimize conflicts here).

Reply:
As an aside, I have no problem if the `GetServiceDefinition` logic is refactored to be criteria-based, it has served its purpose well for years, but can certainly be changed for consistency.

Reply:
@lucasponce are you going to do that refactoring in another PR? If so, we can wait for that PR to go in, then we rebase this PR on that work after it is done.

Reply:
@jmazzitelli yes, I'll prepare one minor refactoring for that. I hope that will help to avoid conflicts with several parallel PRs.